src/iceberg/catalog/rest/resource_paths.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+
+#include "iceberg/catalog/rest/config.h"
+#include "iceberg/catalog/rest/iceberg_rest_export.h"
+#include "iceberg/result.h"
+#include "iceberg/table_identifier.h"
+
+/// \file iceberg/catalog/rest/resource_paths.h
+/// \brief Resource path construction for Iceberg REST API endpoints.
+
+namespace iceberg::rest {
+
+/// \brief Resource path builder for Iceberg REST catalog endpoints.
+///
+/// This class constructs REST API endpoint URLs for various catalog operations.
+class ICEBERG_REST_EXPORT ResourcePaths {
+ public:
+  /// \brief Construct a ResourcePaths with REST catalog configuration.
+  /// \param config The REST catalog configuration containing the base URI
+  static Result<std::unique_ptr<ResourcePaths>> Make(const RestCatalogConfig& config);
+
+  /// \brief Get the /v1/{prefix}/config endpoint path.
+  std::string V1Config() const;
```

Comment:
```suggestion
  std::string Config() const;
```

Let's remove the `V1` prefix. I would expect `ResourcePaths` can handle the version evolution in the future instead of letting caller to handle this.